app/src/main/res/layout/view_legacy_omnibar.xml

```diff
@@ -15,303 +15,300 @@
+                    app:layout_constraintTop_toTopOf="@id/voiceSearchButton" />
 
                 <ImageView
-                    android:id="@+id/fireIconImageView"
+                    android:id="@+id/clearTextButton"
                     android:layout_width="wrap_content"
                     android:layout_height="wrap_content"
-                    android:layout_gravity="center"
-                    android:background="?attr/selectableItemBackgroundBorderless"
-                    android:contentDescription="@string/browserPopupMenu"
+                    android:background="?selectableItemBackgroundBorderless"
+                    android:contentDescription="@string/clearButtonContentDescription"
                     android:padding="@dimen/toolbarIconPadding"
-                    android:src="@drawable/ic_fire" />
-            </FrameLayout>
+                    android:src="@drawable/ic_close_24"
+                    android:visibility="gone"
+                    app:layout_constraintBottom_toBottomOf="parent"
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintTop_toTopOf="parent"
+                    tools:visibility="visible" />
+
+            </androidx.constraintlayout.widget.ConstraintLayout>
+
+        </androidx.appcompat.widget.Toolbar>
 
-            <com.duckduckgo.app.browser.TabSwitcherButton
-                android:id="@+id/tabsMenu"
+        <ProgressBar
+            android:id="@+id/pageLoadingIndicator"
+            style="@style/Widget.AppCompat.ProgressBar.Horizontal"
+            android:layout_width="match_parent"
+            android:layout_height="4dp"
+            android:progressDrawable="@drawable/loading_progress"
+            android:visibility="invisible"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            tools:progress="50"
+            tools:visibility="visible" />
+
+        <FrameLayout
+            android:id="@+id/fireIconMenu"
+            android:layout_width="wrap_content"
+            android:layout_height="match_parent"
+            android:paddingHorizontal="@dimen/keyline_2"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toStartOf="@id/tabsMenu"
+            app:layout_constraintStart_toEndOf="@id/toolbar"
+            app:layout_constraintTop_toTopOf="parent"
+            tools:visibility="visible">
+
+            <ImageView
+                android:id="@+id/fireIconImageView"
                 android:layout_width="wrap_content"
                 android:layout_height="wrap_content"
-                android:layout_marginStart="@dimen/keyline_2"
-                android:layout_marginEnd="@dimen/keyline_2"
-                android:background="?selectableItemBackgroundBorderless"
+                android:layout_gravity="center"
```

Comment:
more padding changes. is this supposed to be here?

Reply:
See my previous comment.